[BUG]  Agent mode does not work on PyCharm 2026.1

## Bug Description

Agent mode does not work on PyCharm 2026.1 using llama3.3 model from Ollama on Ubuntu Linux. Ask mode works fine.

## Steps to Reproduce

1. Curl latest Ollama
2. Install and run the llama3.3 model
3. Install devoxxGenie plugin on pycharm
4. Toggle agent mode, ask any question

## Expected Behavior

Get agentic behaviour

## Actual Behavior

Get crash

## Screenshots/Logs

15:30:18.946 [prompt-exec-4] ERROR c.d.g.s.prompt.error.PromptException - ERROR:Error occurred while processing chat message - false
15:30:18.946 [prompt-exec-4] ERROR c.d.g.s.p.error.PromptErrorHandler - Error occurred while processing chat message
java.util.concurrent.CompletionException: java.lang.NoClassDefFoundError: com/intellij/psi/PsiModifierListOwner
	at java.base/java.util.concurrent.CompletableFuture.wrapInCompletionException(CompletableFuture.java:323)
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:359)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:364)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1791)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.lang.NoClassDefFoundError: com/intellij/psi/PsiModifierListOwner
	at com.devoxx.genie.service.agent.tool.BuiltInToolProvider.registerPsiTools(BuiltInToolProvider.java:416)
	at com.devoxx.genie.service.agent.tool.BuiltInToolProvider.<init>(BuiltInToolProvider.java:205)
	at com.devoxx.genie.service.agent.AgentToolProviderFactory.createToolProvider(AgentToolProviderFactory.java:59)
	at com.devoxx.genie.service.prompt.response.nonstreaming.NonStreamingPromptExecutionService.processChatMessage(NonStreamingPromptExecutionService.java:215)
	at com.devoxx.genie.service.prompt.response.nonstreaming.NonStreamingPromptExecutionService.lambda$executeQuery$0(NonStreamingPromptExecutionService.java:83)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1789)


## Environment

- **PyCharm Version**: 2026.1
- **Plugin Version**: 1.7.4
- **Operating System**: Ubuntu 22.04
- **LLM Provider**: Ollama
- **Model**: llama3.3

--- a/src/main/java/com/devoxx/genie/service/agent/tool/BuiltInToolProvider.java
+++ b/src/main/java/com/devoxx/genie/service/agent/tool/BuiltInToolProvider.java
@@ -337,6 +337,15 @@ public class BuiltInToolProvider implements ToolProvider {
                 new FindImplementationsToolExecutor(project)
         );
 
+        // The remaining tools are Java-only: their executor classes reference Java-plugin PSI
+        // types (PsiMethod, PsiModifierListOwner, …) in method signatures, so merely linking
+        // them throws NoClassDefFoundError in IDEs without the Java plugin (PyCharm, WebStorm,
+        // GoLand, …) — see issue #1100. The isJavaAvailable() checks inside their execute()
+        // methods cannot help because the class never loads. Skip registration entirely.
+        if (!PsiToolUtils.isJavaAvailable()) {
+            return;
+        }
+
         // find_callees — outgoing calls from a method (inverse of find_references)
         tools.put(
                 ToolSpecification.builder()

--- a/src/test/java/com/devoxx/genie/service/agent/tool/BuiltInToolProviderTest.java
+++ b/src/test/java/com/devoxx/genie/service/agent/tool/BuiltInToolProviderTest.java
@@ -1,6 +1,7 @@
 package com.devoxx.genie.service.agent.tool;
 
 import com.devoxx.genie.service.agent.tool.psi.PsiToolCatalog;
+import com.devoxx.genie.service.agent.tool.psi.PsiToolUtils;
 import com.devoxx.genie.ui.settings.DevoxxGenieStateService;
 import com.intellij.openapi.project.Project;
 import dev.langchain4j.agent.tool.ToolSpecification;
@@ -268,6 +269,32 @@ class BuiltInToolProviderTest {
         assertThat(toolNames).contains(
                 "find_symbols", "document_symbols", "find_references", "find_definition",
                 "find_implementations", "find_callees", "trace_call_chains", "calculate_complexity");
+    }
+
+    /**
+     * Regression test for issue #1100: on non-Java IDEs (PyCharm, WebStorm, GoLand…) the
+     * classes of the four Java-only PSI executors cannot even be linked — their method
+     * signatures reference Java-plugin PSI types such as {@code PsiModifierListOwner},
+     * so constructing them throws {@link NoClassDefFoundError} and kills agent mode
+     * entirely. When Java PSI is unavailable, only the five platform-level PSI tools
+     * may be registered; the Java-only executors must never be instantiated.
+     */
+    @Test
+    void provideTools_psiToolsEnabledButJavaPsiUnavailable_registersOnlyPlatformPsiTools() {
+        when(stateService.getPsiToolsEnabled()).thenReturn(true);
+
+        try (MockedStatic<PsiToolUtils> psiToolUtilsMock = mockStatic(PsiToolUtils.class)) {
+            psiToolUtilsMock.when(PsiToolUtils::isJavaAvailable).thenReturn(false);
+            BuiltInToolProvider provider = createProvider();
+
+            Set<String> toolNames = getToolNames(provider.provideTools(request));
+
+            assertThat(toolNames).contains(
+                    "find_symbols", "document_symbols", "find_references",
+                    "find_definition", "find_implementations");
+            assertThat(toolNames).doesNotContain(
+                    "find_callees", "trace_call_chains", "calculate_complexity", "find_dead_code");
+        }
     }
 
     // --- All features enabled ---

